keyboards/thedogkeyboard/thedogkeyboard.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2019 kakunpc
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```

Comment:
If this represents every possible location for a switch on the PCB, it should be named `LAYOUT_all`. In that case, the following locations would need updating as well:

- [info.json, Line 8](https://github.com/qmk/qmk_firmware/blob/f05f650937e99e266032e8ec3939abc3c9a67a9c/keyboards/thedogkeyboard/info.json#L8)
- [keymaps/default/keymap.c, Line 19](https://github.com/qmk/qmk_firmware/blob/f05f650937e99e266032e8ec3939abc3c9a67a9c/keyboards/thedogkeyboard/keymaps/default/keymap.c#L19)
- [keymaps/kakunpc/keymap.c, Line 27](https://github.com/qmk/qmk_firmware/blob/f05f650937e99e266032e8ec3939abc3c9a67a9c/keyboards/thedogkeyboard/keymaps/kakunpc/keymap.c#L27)
- [keymaps/kakunpc/keymap.c, Line 35](https://github.com/qmk/qmk_firmware/blob/f05f650937e99e266032e8ec3939abc3c9a67a9c/keyboards/thedogkeyboard/keymaps/kakunpc/keymap.c#L35)